app/models/pool/candidates.rb

```diff
@@ -0,0 +1,37 @@
+class Pool::Candidates
+  attr_reader :providers
+
+  def initialize(providers:)
+    @providers = providers
+  end
+
+  def self.for_provider(providers:)
+    new(providers:).for_provider
+  end
+
+  def for_provider
+    dismissed_candidates = Candidate.joins(:pool_dismissals).where(pool_dismissals: { provider: providers })
+    rejected_candidate_ids = curated_application_forms.pluck(:candidate_id).uniq
```

Comment:
Small suggestion, `pluck` will execute a query where we don't need to. 
```suggestion
    rejected_candidate_ids = curated_application_forms.select(:candidate_id)
```

How would you feel about moving this all into a method, maybe `reject_candidates_this_cycle`